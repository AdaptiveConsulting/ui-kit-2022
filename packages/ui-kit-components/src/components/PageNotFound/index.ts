export type { Props as PageNotFoundProps } from './PageNotFound';
export { default as PageNotFound } from './PageNotFound';
