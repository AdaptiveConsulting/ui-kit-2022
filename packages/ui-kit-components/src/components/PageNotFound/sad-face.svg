<svg width="256" height="256" viewBox="0 0 256 256" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
    <path d="M127.857 254.307C58.1321 254.307 1.40662 197.581 1.40662 127.857C1.40662 58.1321 58.1321 1.4066 127.857 1.4066C197.581 1.4066 254.307 58.1321 254.307 127.857C254.307 197.581 197.581 254.307 127.857 254.307ZM127.857 12.6466C64.3309 12.6466 12.6466 64.3309 12.6466 127.857C12.6466 191.382 64.3309 243.067 127.857 243.067C191.382 243.067 243.067 191.382 243.067 127.857C243.067 64.3309 191.382 12.6466 127.857 12.6466Z"/>
    <path d="M86.6733 111.384C95.7737 111.384 103.151 104.007 103.151 94.9065C103.151 85.8061 95.7737 78.4287 86.6733 78.4287C77.5728 78.4287 70.1954 85.8061 70.1954 94.9065C70.1954 104.007 77.5728 111.384 86.6733 111.384Z"/>
    <path d="M169.062 111.384C178.163 111.384 185.54 104.007 185.54 94.9065C185.54 85.8061 178.163 78.4287 169.062 78.4287C159.962 78.4287 152.585 85.8061 152.585 94.9065C152.585 104.007 159.962 111.384 169.062 111.384Z"/>
    <path d="M204.235 185.433C202.985 185.433 201.726 185.02 200.681 184.166C180.179 167.416 154.316 158.191 127.857 158.191C101.395 158.191 75.5344 167.416 55.0326 184.166C52.6329 186.133 49.0895 185.774 47.1253 183.371C45.1611 180.968 45.5179 177.428 47.9205 175.464C70.423 157.075 98.8124 146.951 127.857 146.951C156.901 146.951 185.287 157.078 207.793 175.461C210.195 177.425 210.552 180.966 208.588 183.368C207.478 184.728 205.862 185.433 204.235 185.433Z"/>
</svg>
