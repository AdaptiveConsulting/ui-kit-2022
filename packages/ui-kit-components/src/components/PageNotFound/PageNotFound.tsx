import { Box, Paper, Theme, Typography } from '@mui/material';

import { LogoText } from '../Branding/branding';
import { Button } from '../Button';
import {
  default as TextSearch,
  TextSearchOptionBase,
  TextSearchProps,
} from '../TextSearch/TextSearch';
import { ReactComponent as SadFace } from './sad-face.svg';

const styles = {
  svgColor: ({ palette }: Theme) => ({
    color: palette.mode === 'light' ? 'paper.black' : 'paper.white',
  }),
  paper: ({ breakpoints }: Theme) => ({
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    p: 5,
    [breakpoints.down('sm')]: {
      p: 4,
    },
  }),
  container: {
    display: 'flex',
    flexDirection: 'column',
  },
  sadFace: ({ breakpoints }: Theme) => ({
    height: 128,
    width: 'auto',
    alignSelf: 'center',
    [breakpoints.down('sm')]: {
      height: 96,
    },
  }),
  header: {
    textTransform: 'none',
  },
  logo: {
    width: 76,
    height: 'auto',
  },
  logoText: {
    width: 116,
    height: 'auto',
  },
  textSearch: {
    maxWidth: 400,
    mx: 'auto',
    width: '100%',
    p: 2,
  },
  button: {
    '& > .MuiButton-root': { p: 3 },
    textAlign: 'center',
  },
};

export interface Props<T extends TextSearchOptionBase> {
  onNavigateHome: (e: React.BaseSyntheticEvent) => void;
  fitContainer: boolean;
  inputProps: TextSearchProps<T>;
}

const PageNotFound = <T extends TextSearchOptionBase>({
  onNavigateHome,
  fitContainer,
  inputProps,
}: Props<T>) => (
  <Paper
    sx={[
      {
        minWidth: `100${fitContainer ? '%' : 'vw'}`,
        minHeight: `100${fitContainer ? '%' : 'vh'}`,
      },
      styles.paper,
    ]}
    elevation={0}
    square={true}
  >
    <Box
      sx={[styles.container, { maxHeight: '646px' }]}
      flex="1 1 0"
      justifyContent="center"
      gap={6}
    >
      <SadFace sx={[styles.sadFace, styles.svgColor]} />
      <Box sx={styles.container} gap={2}>
        <Typography variant="h4" sx={styles.header} align="center">
          We couldn&apos;t find this page.
        </Typography>
        <Typography variant="body2" align="center">
          Let’s help you find what you’re looking for.
        </Typography>
      </Box>
      <Box sx={styles.textSearch}>
        <TextSearch {...inputProps} fullWidth={true} />
      </Box>
      <Box sx={styles.button}>
        <Button variant="SECONDARY" click={onNavigateHome}>
          Take Me Back
        </Button>
      </Box>
    </Box>
    <Box sx={styles.container} justifyContent="flex-end">
      <Box alignSelf="flex-end">
        <LogoText sx={[styles.logoText, styles.svgColor]} />
        <Typography variant="body2" sx={styles.logoText}>
          © 2022 Reactive Analytics
        </Typography>
      </Box>
    </Box>
  </Paper>
);

export default PageNotFound;
