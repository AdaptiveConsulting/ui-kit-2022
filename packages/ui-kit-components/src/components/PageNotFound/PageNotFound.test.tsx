import * as React from 'react';
import '@testing-library/jest-dom';

import { fireEvent, render, screen } from '@testing-library/react';

import PageNotFound from './PageNotFound';

jest.mock('./sad-face.svg', () => ({
  __esModule: true,
  ReactComponent: () => {
    return <p>Sad Face</p>;
  },
}));

jest.mock('../TextSearch/TextSearch', () => ({
  __esModule: true,
  default: ({ ...inputProps }) => {
    return <p>{inputProps.placeholder}</p>;
  },
}));

jest.mock('../Branding/branding', () => ({
  LogoText: () => {
    return <p>Logo Text</p>;
  },
}));

jest.mock('../Button', () => ({
  __esModule: true,
  Button: ({ click }: { click: (e: React.BaseSyntheticEvent) => void }) => {
    return <button onClick={click}>Test Button</button>;
  },
}));

describe('PageNotFound coponent tests', () => {
  describe('Given render a simple component', () => {
    beforeEach(() => {
      render(
        <PageNotFound
          onNavigateHome={() => null}
          fitContainer={true}
          inputProps={{ options: [] }}
        />,
      );
    });

    it('Then get the same snapshot', () => {
      expect(screen.getByText('© 2022 Reactive Analytics')).toMatchSnapshot();
    });

    it('Then get the right company info', () => {
      expect(screen.getByText('© 2022 Reactive Analytics')).toBeInTheDocument();
    });
  });

  describe('Given render a component with custom props', () => {
    const clickFn = jest.fn();
    beforeEach(() => {
      render(
        <PageNotFound
          onNavigateHome={clickFn}
          fitContainer={true}
          inputProps={{ options: [], placeholder: 'Place Holder' }}
        />,
      );
    });

    it('Then should display the right placeholder value', () => {
      expect(screen.getByText('Place Holder')).toMatchSnapshot();
    });

    describe('When click the test button', () => {
      beforeEach(() => {
        fireEvent.click(screen.getByText('Test Button'));
      });
      it('Then should trigger the click button function', () => {
        expect(clickFn).toBeCalled();
      });
    });
  });
});
