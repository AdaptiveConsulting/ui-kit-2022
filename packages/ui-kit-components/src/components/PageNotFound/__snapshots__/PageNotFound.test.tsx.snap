// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound coponent tests Given render a component with custom props Then should display the right placeholder value 1`] = `
<p>
  Place Holder
</p>
`;

exports[`PageNotFound coponent tests Given render a simple component Then get the same snapshot 1`] = `
<p
  class="MuiTypography-root MuiTypography-body2 css-up49h5-MuiTypography-root"
>
  Â© 2022 Reactive Analytics
</p>
`;
